src/atopile/front_end.py

```diff
@@ -283,22 +283,61 @@ def visitSimple_stmt(self, ctx: ap.Simple_stmtContext | Any) -> KeyOptMap:
             return super().visitChildren(ctx)
         return KeyOptMap.empty()
 
+    # TODO: @v0.4: remove this shimming
+    @staticmethod
+    def _find_shim(file_path: Path | None, ref: Ref) -> tuple[Type[L.Node], str] | None:
+        if file_path is None:
+            return None
+
+        import_addr = address.AddrStr.from_parts(file_path, str(Ref(ref)))
+
+        for shim_addr in shim_map:
+            if import_addr.endswith(shim_addr):
+                return shim_map[shim_addr]
+
+        return None
+
     @classmethod
     def survey(
         cls, file_path: Path | None, ctx: ap.BlockdefContext | ap.File_inputContext
     ) -> Context:
         surveyor = cls()
         context = Context(file_path=file_path, scope_ctx=ctx, refs={})
         for ref, (item, item_ctx) in surveyor.visit(ctx):
+            assert isinstance(item_ctx, ParserRuleContext)
             if ref in context.refs:
                 # Downgrade the error in case we're shadowing things
+                # Not limiting the number of times we show this warning
+                # because they're pretty important and Wendy is well cached
                 with downgrade(errors.UserKeyError):
                     raise errors.UserKeyError.from_ctx(
                         item_ctx,
                         f'"{ref}" already declared. Shadowing original.'
                         " In the future this may be an error",
                     )
-            context.refs[ref] = item
+
+            # TODO: @v0.4: remove this shimming
+            if shim := cls._find_shim(context.file_path, ref):
+                shim_cls, preferred = shim
+
+                if hasattr(item_ctx, "name"):
+                    dep_ctx = item_ctx.name()  # type: ignore
+                elif hasattr(item_ctx, "name_or_attr"):
+                    dep_ctx = item_ctx.name_or_attr()  # type: ignore
+                else:
+                    dep_ctx = item_ctx
+
+                with downgrade(DeprecatedException):
+                    raise DeprecatedException.from_ctx(
+                        dep_ctx,
+                        "Is deprecated and will be removed in a future"
+                        f" version. Use {preferred} instead.",
+                    )
+
+                context.refs[ref] = shim_cls
```

Comment:
This shims in the new components at a lower-level, so they work within the std lib as well